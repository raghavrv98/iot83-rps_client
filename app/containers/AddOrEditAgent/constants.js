/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
