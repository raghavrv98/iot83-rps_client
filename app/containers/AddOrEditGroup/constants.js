/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
