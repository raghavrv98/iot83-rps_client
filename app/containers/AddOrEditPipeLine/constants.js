/*
 *
 * AddOrEditPipeLine constants
 *
 */

export const DEFAULT_ACTION = "app/AddOrEditPipeLine/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddOrEditPipeLine/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditPipeLine/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditPipeLine/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PIPELINE_REQUEST = "app/AddOrEditPipeLine/GET_PIPELINE_REQUEST";
export const GET_PIPELINE_REQUEST_SUCCESS = "app/AddOrEditPipeLine/GET_PIPELINE_REQUEST_SUCCESS";
export const GET_PIPELINE_REQUEST_FAILURE = "app/AddOrEditPipeLine/GET_PIPELINE_REQUEST_FAILURE";
