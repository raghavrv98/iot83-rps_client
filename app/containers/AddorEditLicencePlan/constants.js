/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";