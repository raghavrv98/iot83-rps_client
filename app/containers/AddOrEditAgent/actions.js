/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
