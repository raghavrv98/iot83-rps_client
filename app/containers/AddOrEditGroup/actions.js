/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
