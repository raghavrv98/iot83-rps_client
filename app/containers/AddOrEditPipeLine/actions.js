/*
 *
 * AddOrEditPipeLine actions
 *
 */

import { ON_SUBMIT_REQUEST, GET_PIPELINE_REQUEST } from "./constants";

export function onSubmitHandler(payload, id, pipelineId) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    pipelineId,
    id
  };
}

export function getPipelineDetails(id, pipelineId) {
  return {
    type: GET_PIPELINE_REQUEST,
    id,
    pipelineId
  };
}

