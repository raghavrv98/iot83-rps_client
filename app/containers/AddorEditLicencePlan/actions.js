/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}