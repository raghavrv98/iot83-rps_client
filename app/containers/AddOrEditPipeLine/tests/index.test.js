import React from "react";
import { shallow} from "enzyme";
import history from "utils/history";
import { AddOrEditPipeLine, mapDispatchToProps } from "../index";
import { ON_SUBMIT_REQUEST, GET_PIPELINE_REQUEST } from "../constants";
import { onSubmitHandler,getPipelineDetails } from "../actions";

const match = {
  params: {
    id: ""
  }
};
const submitError = "this is submit success";
const submitSuccess = "this is submit error";
const pipelineDetailsSuccess = "this is pipeline details success";
const pipelineDetailsFailure = "this is pipeline details Error";

describe("<AddOrEditPipeLine />", () => {
  // test("renders clicking back button trigger onClick event error", () => {
  //   let expectedCount = history.length + 1;
  //   const resetToInitailState = jest.fn();
  //   const wrapper = shallow(
  //     <AddOrEditPipeLine
  //       history={history}
  //       match={match}
  //       resetToInitailState={resetToInitailState}
  //     />
  //   );
  //   let backbutton = wrapper.find("button.btn-transparent");
  //   expect(backbutton.length).toBe(1);
  //   backbutton.simulate("click");
  //   expect(history.length + 1).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/");
  // });

  it("clicking submit button trigger onClick event", async () => {
    const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} onSubmitHandler={onSubmitHandler} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'formSubmitHandler');
    wrapper.instance().forceUpdate(); 
    let addOrEditForm = wrapper.find("form.contentForm");
    expect(addOrEditForm.length).toBe(1);
    addOrEditForm.simulate('submit', {preventDefault: () => {
    }});
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  });

  it("clicking submit button bad low band trigger onClick event", async () => {
    const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} onSubmitHandler={onSubmitHandler} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'formSubmitHandler');
    wrapper.instance().forceUpdate();
    let addOrEditForm = wrapper.find("form.contentForm");
    expect(addOrEditForm.length).toBe(1);
    addOrEditForm.simulate('submit', {preventDefault: () => {
    }});
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  });

// it("clicking onChange button trigger onClick event", async () => {
//   const resetToInitailState = jest.fn();
//   const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} getPipelineDetails={getPipelineDetails} />);
//   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'nameChangeHandler');
//   wrapper.instance().forceUpdate(); 
//   wrapper.find('[id="name"]').simulate('change',{target: {id: "name", value: "name"}});
//   expect(wrapper.find('[id="name"]').props().value).toEqual("name");
//   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
// });
// it("clicking onChange button trigger onClick event", async () => {
//   const resetToInitailState = jest.fn();
//   const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} getPipelineDetails={getPipelineDetails} resetToInitailState={resetToInitailState} />);
//   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'nameChangeHandler');
//   wrapper.instance().forceUpdate(); 
//   wrapper.find('[id="processFluid"]').simulate('change',{target: {id: "processFluid", value: "processFluid"}});
//   expect(wrapper.find('[id="processFluid"]').props().value).toEqual("processFluid");
//   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
// });
// it("clicking onChange button trigger onClick event", async () => {
//   const resetToInitailState = jest.fn();
//   const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} getPipelineDetails={getPipelineDetails} resetToInitailState={resetToInitailState} />);
//   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'nameChangeHandler');
//   wrapper.instance().forceUpdate(); 
//   wrapper.find('[id="overallGeometry"]').simulate('change',{target: {id: "overallGeometry", value: "overallGeometry"}});
//   expect(wrapper.find('[id="overallGeometry"]').props().value).toEqual("overallGeometry");
//   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
// });
// it("clicking onChange button trigger onClick event", async () => {
//   const resetToInitailState = jest.fn();
//   const wrapper = shallow(<AddOrEditPipeLine history={history} match={match} getPipelineDetails={getPipelineDetails} resetToInitailState={resetToInitailState} />);
//   const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'nameChangeHandler');
//   wrapper.instance().forceUpdate(); 
//   wrapper.find('[id="other"]').simulate('change',{target: {id: "other", value: "other"}});
//   expect(wrapper.find('[id="other"]').props().value).toEqual("other");
//   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
// });
  it("test component will receive props", async () => {
    // const resetToInitailState = jest.fn();
    // let expectedCount = history.length + 1;
    spyOn(
      AddOrEditPipeLine.prototype,
      "componentWillReceiveProps"
    ).and.callThrough();
    const wrapper = shallow(
      <AddOrEditPipeLine
        history={history}
        match={match}
        onSubmitHandler={onSubmitHandler}
        submitSuccess=""
        submitError=""
        pipelineDetailsSuccess=""
        pipelineDetailsFailure=""
      />
    );
    wrapper.setProps({
      submitError,
      submitSuccess,
      pipelineDetailsSuccess,
      pipelineDetailsFailure
    });
    expect(
      AddOrEditPipeLine.prototype.componentWillReceiveProps
    ).toHaveBeenCalledTimes(1);
  });
  it("test edit case", async () => {
    const mockParams = {
      params: {
        id: "demoId",
        pipelineId:'demoId'
      }
    };

    spyOn(AddOrEditPipeLine.prototype, "componentDidMount").and.callThrough();
    const wrapper = shallow(
      <AddOrEditPipeLine
        history={history}
        match={mockParams}
        onSubmitHandler={onSubmitHandler}
        // resetToInitailState={resetToInitailState}
        getPipelineDetails={getPipelineDetails}
      />
    );
    expect(AddOrEditPipeLine.prototype.componentDidMount).toHaveBeenCalledTimes(1);
  });

  // it("Input should call handleChange on name change event", () => {
  //   const resetToInitailState = jest.fn();
  //   const inputChangeHandlerSpy = jest.spyOn(AddOrEditPipeLine.propTypes, "nameChangeHandler");
  //    const wrapper = shallow(<AddOrEditPipeLine history={history} match={match}  getPipelineDetails={getPipelineDetails} /> );
  //   wrapper.find('[id="name"]').simulate("change", { target: { id: "name", value: "name" } });
  //   expect(wrapper.find('[id="name"]').props().value).toEqual("name");
  //   expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  //  });

  it("Function testing on mapDispatchToProps", () => {
    const dispatch = jest.fn();
    const payload = "demo";
    const pipelineId = "demoId";
    const id = "demoId";
  
    mapDispatchToProps(dispatch).onSubmitHandler(payload, pipelineId, id);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: ON_SUBMIT_REQUEST,
      payload: "demo",
      pipelineId:"demoId",
      id: "demoId"
    });
    mapDispatchToProps(dispatch).getPipelineDetails(pipelineId, id);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: GET_PIPELINE_REQUEST,
      id: "demoId",
      pipelineId:"demoId"
    });
  });

  it("match snapshots", () => {
    // const resetToInitailState = jest.fn();
    const wrap = shallow(
      <AddOrEditPipeLine
        history={history}
        match={match}
        onSubmitHandler={onSubmitHandler}
      />
    );
    expect(wrap).toMatchSnapshot();
  });
});

// /**
//  * Factory function to create a shallowWrapper for the App component.
//  * @function setup
//  * @param {object} props - component props specific to this setup.
//  * @param {any} state - Initial state for setup
//  * @returns {shallowWrapper}
//  */

// const setup = (props = {}, state = null) =>{
//     const wrapper = shallow(<AddOrEditPipeLine { ...props } />)
//     if(state) wrapper.setState(state);
//     return wrapper;
// }

// /**
//  * Return ShallowWrapper containg node(s) with the given data-test value.
//  * @param {shallowWrapper} wrapper - Enzyme shallow wrapper to search within.
//  * @param {string} val - value of data-test attribute for search.
//  * @returns {shallowWrapper}
//  */

// const findByTestAttr = (wrapper, val) => {
//     return wrapper.find(`[data-test="${val}"]`);
// }