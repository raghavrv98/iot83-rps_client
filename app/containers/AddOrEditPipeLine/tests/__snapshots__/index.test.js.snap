// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddOrEditPipeLine /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      AddOrEditPipeLine
    </title>
    <meta
      content="Description of AddOrEditPipeLine"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Pipelines"
            id="app.common.messages.managePipeline"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Create Pipeline"
          id="app.containers.AddOrEditPipeLine.addnewpipeline"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
  </div>
  <form
    className="contentForm"
    onSubmit={[Function]}
  >
    <h5
      className="formHeader"
    >
      <FormattedMessage
        defaultMessage="Create Pipeline"
        id="app.containers.AddOrEditPipeLine.addnewpipeline"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
      <div
        className="headerItemRight"
      >
        <div
          className="switchButtonLabel"
        >
          <label
            className="switchLabelText"
          >
            <FormattedMessage
              defaultMessage="Generate Report"
              id="app.containers.AddOrEditPipeLine.generateReport"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
             :
          </label>
          <label
            className="switchLabel"
          >
            <input
              checked={false}
              id="generateReport"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="switchMark"
            />
          </label>
        </div>
      </div>
    </h5>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Pipeline Name:"
          id="app.containers.AddOrEditPipeLine.pipelinename"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Process Fluid"
          id="app.common.messages.processfluid"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        :
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        className="form-control"
        id="processFluid"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Hot HH Limit"
          id="app.common.messages.hotHHLimit"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="Hot_HH_Limit"
          max="250.0"
          min="0"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            ℃
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Cold LL Limit"
          id="app.common.messages.coldLLLimit"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="Cold_LL_Limit"
          max="250.0"
          min="-120.0"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            ℃
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Max Alarm Distance"
          id="app.common.messages.maxAlarmDistance"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="maxAlarmDistance"
          min="1"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            <FormattedMessage
              defaultMessage="m"
              id="app.common.messages.m"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Freeze Band Low Limit"
          id="app.common.messages.freezeBandLowLimit"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="FreezeBandLowLimit"
          max="250.0"
          min="-120.0"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            ℃
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Freeze Temperature"
          id="app.common.messages.freezeTemperature"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="Freeze_Temperature"
          max="250.0"
          min="-120.0"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            ℃
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Freeze Band High Limit"
          id="app.common.messages.freezeBandHighLimit"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="FreezeBandHighLimit"
          max="250.0"
          min="-120.0"
          onChange={[Function]}
          required={true}
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            ℃
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group position-relative"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Time To Freeze LL Limit"
          id="app.common.messages.timeToFreezeLLLimit"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          id="TimeToFreeze_LL_Limit"
          max="16.0"
          min="2.0"
          onChange={[Function]}
          required={true}
          step="any"
          type="number"
          value=""
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
          >
            <FormattedMessage
              defaultMessage="Hours"
              id="app.common.messages.hours"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Anchor Shift Threshold"
          id="app.containers.AddOrEditPipeLine.anchorShiftThreshold"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        className="form-control"
        id="anchorShiftThreshold"
        onChange={[Function]}
        required={true}
        step="any"
        type="number"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Geometry:"
          id="app.containers.AddOrEditPipeLine.geometry"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <textarea
        className="form-control"
        id="overallGeometry"
        onChange={[Function]}
        required={true}
        rows="3"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Other Info:"
          id="app.containers.AddOrEditPipeLine.otherinfo"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </label>
      <textarea
        className="form-control"
        id="other"
        onChange={[Function]}
        rows="3"
        type="text"
        value=""
      />
    </div>
    <div
      className="flex"
    >
      <div
        className="fx-b20"
      />
      <div
        className="fx-b80"
      >
        <p
          className="note"
        >
          <span
            className="noteText"
          >
             
            <FormattedMessage
              defaultMessage="Note"
              id="app.common.messages.note"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
             : 
          </span>
          <b>
            <FormattedMessage
              defaultMessage="Cold LL Limit should be smaller than Hot HH Limit"
              id="app.containers.AddOrEditPipeLine.note1Bold"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </b>
           
          <FormattedMessage
            defaultMessage="( Cold LL Limit should be between -121 and 251 and Hot HH Limit should be between -1 and 251 )."
            id="app.containers.AddOrEditPipeLine.note1"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </p>
        <p
          className="note pd-l-40"
        >
          <b>
            <FormattedMessage
              defaultMessage="Freeze Band Low Limit should be less than Freeze Temperature"
              id="app.containers.AddOrEditPipeLine.note2Bold"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </b>
           
          <FormattedMessage
            defaultMessage="( Freeze Temperature should be between -121 and 251 and Time To Freeze LL Limit should be between 1 and 17 )."
            id="app.containers.AddOrEditPipeLine.note2"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </p>
      </div>
    </div>
    <div
      className="form-group justify-content-end mt-3"
    >
      <button
        className="btn btn-danger"
        id="savePipeline"
        type="submit"
      >
        <i
          className="far fa-check-circle"
        />
        <FormattedMessage
          defaultMessage="Save"
          id="app.common.messages.save"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </div>
  </form>
</div>
`;
